lib/ddtrace/contrib/mongodb/subscribers.rb

```diff
@@ -0,0 +1,104 @@
+module Datadog
+  module Contrib
+    # MongoDB module includes classes and functions to instrument MongoDB clients
+    module MongoDB
+      # `MongoCommandSubscriber` listens to all events from the `Monitoring`
+      # system available in the Mongo driver.
+      class MongoCommandSubscriber
+        # rubocop:disable Metrics/MethodLength
+        # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
The only one that matters to me is the cyclo test, which is triggered by the `switch/case + nested` `unless` calls I think. Technically you could write 5 liners for all those case sections but I'm not dead sure it would make the code more readable. I'm fine with leaving the warning here.